File: 2026.2 Estrutura de dados/Exercicios/0.4 - gorjeta.c
#include <stdio.h>
void calcularGorjeta(double valor, double taxa) {
    if (valor > 0) {
        double taxaDeGarcom = valor * (taxa / 100);
        printf("\n\n******Conta restaurante******");
        printf("consumo restaurante: %.2f\n", valor);
        printf("taxa: %.2f\n", taxaDeGarcom);
        printf("total: %.2f\n\n", taxaDeGarcom + valor);
    } else {
    printf("O valor informado deve ser maior que zero\n\n");
    }
}

int main() {
    printf("Bem vindo ao sistema de definição de gorjeta\n");
    while(1) {
        double valor;
        printf("Digite o valor de consumo no restaurante ou aperte CTRL + C para SAIR!!!\n");
        scanf("%lf", &valor);
        calcularGorjeta(valor, 5);
    }   
}

File: 2026.2 Estrutura de dados/Exercicios/Ray/0.4 - gorjeta.py
def calcularGorjeta(valor: float, taxa: float):
    if (valor > 0):
        taxaDeGarcom = valor * (taxa/100)
        print(f"""\n\n{"*" * 20} Conta restaurante {"*" * 20}
consumo no restaurante: {valor:.2f}
taxa de garçom: {taxaDeGarcom:.2f}
total: {valor + taxaDeGarcom:.2f}\n\n""")
    else:
        print("O valor informado deve ser maior do que 0")




if __name__ == "__main__":
    print("Bem vindo ao sistema de definição gorjeta\n")
    while(True):
        try:
            valorConta = float(input("Digite o valor consumido no restaurante ao total ou CTRL + C para SAIR!!\n"))
            calcularGorjeta(valorConta, 5)
        except KeyboardInterrupt:
            print("Obrigado por utilizar o sistema!\n")
            break
        except Exception:
            print("erro de digitação\n")
            continue